posthog/utils.py

```diff
@@ -299,8 +299,16 @@ def render_template(template_name: str, request: HttpRequest, context: Dict = {}
     posthog_app_context: Dict[str, Any] = {
         "persisted_feature_flags": settings.PERSISTED_FEATURE_FLAGS,
         "anonymous": not request.user or not request.user.is_authenticated,
+        "week_start": 1,  # Monday
     }
 
+    from posthog.api.geoip import get_geoip_properties  # avoids circular import
+
+    geoip_properties = get_geoip_properties(get_ip_address(request))
+    country_code = geoip_properties.get("$geoip_country_code", None)
+    if country_code:
+        posthog_app_context["week_start"] = get_week_start_for_country_code(country_code)
+
```

Comment:
Why not use the browser locale for this? If I visit the US, I won't suddenly change my date habits for a few weeks. Browser/OS locale features have been specifically designed for storing those preferences. This would also avoid the need for using GeoIP in all Django requests.

Reply:
Perhaps we should rethink this.

There's also this issue: https://github.com/PostHog/posthog/issues/8353

Should we make "what is weekly" configurable in trends? Does a week start on a Monday or on a Sunday when you're looking at a line chart? Whatever we decide upon here should likely be configurable in the project settings. Perhaps with GeoIP-based defaults. 

... and if we make that configurable, it makes sense that the calendars follow the same pattern, because, well, consistency.

Reply:
Regarding the questions:

> This would also avoid the need for using GeoIP in all Django requests.

This is normally just one request when you open the app. Not every API call everywhere.

> Why not use the browser locale (option 2) for this?

Experience has shown this to not be very reliable. It's often just "en" or "fr" without much context. Is "en" then "Americans who forgot to specify" or "rest of the world". 

> If I visit the US (or an American Europe), I won't suddenly change my date habits for a few weeks.

I'd try to adopt the local customs as much as possible... and having calendars acting weird when you're in a funny place is part of the fun. 😅

Reply:
Yeah, I see the difficulty. My main concern is just that as a user, I can easily change my language settings, but I can't change my IP. So e.g. if an American expat user asks us if we can show Sunday first, our only answer would be "…well you can always fly back to the US".
Now that I think about this, we could use PostHog to solve this… We should just make `posthog-js` start tracking `navigator.language`. Then we'll be able to see how many Americans have `en` and now many `en-us`.

Reply:
`posthog-js` PR: https://github.com/PostHog/posthog-js/pull/460